### ISSUE TITLE
Contact

### ISSUE THREAD

**Issue by Vash1** [Klexus1/Whys#1]
Hi there, sorry but there was no other way to contact you.
I have prepared a solution you might like on your Github question without heavy loops. It was about counting user emails, but maybe you have already resolved it by yourself?

**Comment by Klexus1** [comment]
Oh I very much appreciate you are reaching out (even this way!) That is awesome. When I read about union() more I found out that it is not the way to go, and furthermore FE guys requested a different format so I had to change it overall but I will very much like to see your suggestion as it hasn't been completed yet. (I deleted the SO question, could you post it as a comment here?) :)

**Comment by Vash1** [comment]
`#event__in just takes a list of pk's but you can also use complex Q filter here
users = User.objects.filter(event__in=[1,2,3])
#the approach is to annotate the count, but... this will not work if you use values("email", "first_name", ...)
mail_count = users.values("email").annotate(Count('email'))
#mail_count = users.values("email").annotate(count=Count('email')) #if you don't like email__count you can also use a named annotation, I like to use get_xxx

pairs = {}
for pair in users.order_by("email__count"): 
    pairs = {**pairs, pair.get("email"): pair.get('email__count')}

for user in users: 
    print(user, pairs.get(user.name_first) 
`
pairs gets appended with {mail: count} pairs in each iteration. I'm not sure if there is a cleaner way to do it, but it's the only loop, while the count itself is performed by the database.
I had minor issues with the resulting loop as I tried this on user-entered data, because of special characters.

Maybe you could also Count the user from the event queryset

**Comment by Klexus1** [comment]
Fair enough, however, I see there the shortcoming that I faced too. when you do
`users = User.objects.filter(event__in=[1,2,3])` this will implicitly call distinct and you will end up with all users, but with a maximum of one occurrence I believe ..
